Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Dashboard/Ride.hs

```diff
@@ -94,6 +100,15 @@ rideEnd merchantShortId reqRideId Common.EndRideReq {point} = withFlowHandlerAPI
   shandle <- EHandler.buildEndRideHandle merchantId
   EHandler.dashboardEndRide shandle rideId dashboardReq
 
+getPointByTryingToRemoveNull :: EsqDBReplicaFlow m r => Maybe LatLong -> Id Common.Ride -> m (Maybe LatLong)
+getPointByTryingToRemoveNull (Just pt) _ = pure $ Just pt
+getPointByTryingToRemoveNull Nothing reqRideId = do
+  let rideId = cast @Common.Ride @DRide.Ride reqRideId
+  ride <- QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
+  booking <- QBooking.findById ride.bookingId >>= fromMaybeM (BookingDoesNotExist ride.bookingId.getId)
+  let bookingLocation = booking.toLocation
+  pure $ Just LatLong {lat = bookingLocation.lat, lon = bookingLocation.lon}
```

Comment:
Your solution requires redundant DB calls.
We already have inside of handler:

```
      case dashboardReq.point of
        Just point -> pure point
        Nothing -> do
          driverLocation <- findDriverLoc booking.providerId driverId >>= fromMaybeM LocationNotFound
          pure $ getCoordinates driverLocation
```

So if we want to change it to booking location, then it should be changed like this:
`      pure $ fromMaybe (LatLong {lat = booking.toLocation.lat, lon = booking.toLocation.lon}) dashboardReq.point`

and `findDriverLoc` should be removed from handler